esmvaltool/cmor/check.py

```diff
@@ -256,6 +256,18 @@ def _check_dim_names(self):
                 except iris.exceptions.CoordinateNotFoundError:
                     try:
                         coord = self._cube.coord(coordinate.standard_name)
+                        if self._cmor_var.table_type == 'CMIP6' and \
+                           coord.var_name in ['latitude', 'longitude']:
+                            # Add comment about dealing with a native grid?
+                            continue
+                        else:
+                            self.report_error(
+                                'Coordinate {0} has var name {1}' \
+                                'instead of {2}',
+                                coordinate.name,
+                                coord.var_name,
+                                coordinate.out_name,
+                                )
                     except iris.exceptions.CoordinateNotFoundError:
```

Comment:
@sloosvel careful here - it may be that `coord` doesn't actually have `var_name` attribute and this will fail with `AttributeError` error - maybe place a check on that first?

Reply:
Do you have an example for that? In the netcdf file that cannot happen, right? So the only possibility could be a badly initialized cube in memory, but even then the worst that could happen is a `var_name` that is `None`, no?

Reply:
Lines 264-270 are how the check was working until now, and coord.var_name is being used there. So I didn't think that would be a problem. I tried to keep everything as it was working, but in the case of latitude and longitude for CMIP6 skip the error message.

Reply:
I think I may have talked out of my arse here, I have seen a lot of cases that `var_name` was not specified in the `DimCoord` object and I assumed it may be optional, it might have been that in those cases it'd be there but be `None` and not shown explicitly; in any case, if it's `None` how do we handle that? - throw the error? seems a bit unjust for those cases, `None` is much better than something like `my_dog_latitude` :beer:

Reply:
At least my_dog_latitude is more specific than None. I thought that the point of the checker was to check against the tables to make sure that the standards are being followed, and if not implement the corresponding fixes? So if any of the coords had var_name as None and that disagrees with the out_name in the tables, why is throwing the error not a good thing?

Reply:
am all OK with what you guys decide - am as impartial to CMOR stuff as a dog is to a cat :grin:

Reply:
OK. I have a couple of suggestions to make life easier for diagnostics developers:

- Allow any var_name for latitude and longitude in CMIP6 only if the coordinate is multidimensional

- Rename var_name to lat lon to make it consistent with 1D case and print a warning explaining why we do it. I am a bit worried about letting pass coordinates with different var_names because is the main reference name to use when using the netcdf library. I am a bit worried about the impact in diagnostics if users are not using things like iris or for other languages

If you are OK with both, we can implement this quickly and finally merge

Reply:
absolutely, that's what @zklaus and @sloosvel were saying too, I believe! welcome back to the grind, bro :beer:

Reply:
Hi @jvegasbsc , good to have you back! Let's continue the general discussion in the thread below.